src/TpetraLinearSystem.C

```diff
@@ -2271,7 +2264,16 @@ int getDofStatus_impl(stk::mesh::Entity node, const Realm& realm)
       }
     }
   }
-
+  
+  // finally, if non conformal is active
+  if ( nonConformalActive ) {
+    if (entityIsOwned)
+      return DS_OwnedDOF;
+    if (!entityIsOwned && (entityIsGhosted || entityIsShared)){
```

Comment:
I'm still not sure this is correct either...

Reply:
This is why dofStatus is a mess. In STK, we have clean definitions of owned, shared and ghosted.  However, dofStatus does not respect this. For the non-conformal all to work properly, a ghosted node due to the non-conformal all needs to be listed as "sharedNotOwned". Every part of the code in Tpetra assumes this is the case.

Sadly, in that symmetry bc example, the ghosted node due to non-conformal was "correct" in listing it as "shared not owned".

I guess that in dofStatus, we have taken over the definition of shared. It means not owned, however, ghosted or shared when the non-conformal all is activated.